lib/driverProviders/attachSession.js

```diff
@@ -0,0 +1,53 @@
+/*
+ *  This is an implementation of the Attach Session Driver Provider.
+ *  It is responsible for setting up the account object, tearing
+ *  it down, and setting up the driver correctly.
+ */
+
+
+var util = require('util'),
+    q = require('q'),
+    DriverProvider = require('./driverProvider'),
+    log = require('../logger'),
+    webdriver = require('selenium-webdriver'),
+    executors = require('selenium-webdriver/executors');
+
+
+var AttachedSessionDriverProvider = function(config) {
+  DriverProvider.call(this, config);
+};
+util.inherits(AttachedSessionDriverProvider, DriverProvider);
+
+/**
+ * Configure and launch (if applicable) the object's environment.
+ * @public
+ * @return {q.promise} A promise which will resolve when the environment is
+ *     ready to test.
+ */
+AttachedSessionDriverProvider.prototype.setupEnv = function() {
+  log.puts('Using the selenium server at ' + this.config_.seleniumAddress);
+  log.puts('Using session id - ' + this.config_.sessionId);
+  return q.fcall(function() {});
+};
+
+
+/**
+ * Getting a new driver by attaching an existing session
+ *
+ * @public
+ * @return webdriver instance
+ */
+AttachedSessionDriverProvider.prototype.getNewDriver = function() {
+  var executor = executors.createExecutor(this.config_.seleniumAddress, null);
+  var newDriver = new webdriver.WebDriver
+      .attachToSession(executor, this.config_.sessionId);
+  this.drivers_.push(newDriver);
+  return newDriver;
+
+};
+
+
```

Comment:
I think we'll need to override `quitDriver` as well to avoid killing the session when we're done with it.